synapseclient/client.py

```diff
@@ -1988,78 +1996,62 @@ def _deleteFileHandle(self, fileHandle):
     ############################################################
 
     def _getDefaultUploadDestination(self, entity):
-        return self.restGET('/entity/%s/uploadDestination'% id_of(entity),
+        return self.restGET('/entity/%s/uploadDestination'% entity['parentId'],
                      endpoint=self.fileHandleEndpoint)
 
 
-    def __getStorageLocation(self, entity):
-        storageLocations = self.restGET('/entity/%s/uploadDestinations'% entity['parentId'],
-                     endpoint=self.fileHandleEndpoint)['list']
-        return storageLocations[0]
-
-        # if uploadHost is None:
-        #     return storageLocations[0]
-        # locations = [l.get('url', 'S3') for l in storageLocations]
-        # uploadHost = entity.get('uploadHost', None)
-
-        # for location in storageLocations:
-        #     #location can either be of  uploadType S3 or SFTP where the latter has a URL
-        #     if location['uploadType'] == 'S3' and uploadHost == 'S3':
-        #         return location
-        #     elif (location['uploadType'] == 'SFTP' and uploadHost != 'S3' and
-        #           utils.is_same_base_url(uploadHost, location['url'])):
-        #         return location
-        # raise SynapseError('You are uploading to a project that supports multiple storage '
-        #                    'locations but have specified the location of %s which is not '
-        #                    'supported by this project.  Please choose one of:\n %s'
-        #                    %(uploadHost, '\n\t'.join(locations)))
-
-
     def __uploadExternallyStoringProjects(self, entity, local_state):
         """Determines the upload location of the file based on project settings and if it is
         an external location performs upload and returns the new url and sets synapseStore=False.
         It not an external storage location returns the original path.
 
         :param entity: An entity with path.
 
-        :returns: A URL or local file path to add to Synapse along with an update local_state
-                  containing externalURL and content-type
+        :returns: A URL or local file path to add to Synapse,
+                  an update local_state containing externalURL and content-type,
+                  and a storage location id indicating to where the entity should be uploaded ("None" defaults to Synapse)
         """
```

Comment:
This is insconsistent with the function header.  The local_state is not returned but is a parameter.

Since we are changing this function we can advantage of the refactor to rename it _uploadExternalllyStoringProjects (_ _ functions are generally a bad idea).